shared/timezones/pyproject.toml

```
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[project]
+name = "apache-airflow-shared-timezones"
+description = "Shared timezone code for Airflow distributions"
+version = "0.0"
+classifiers = [
+    "Private :: Do Not Upload",
+]
+
+dependencies = [
+    "pendulum",
+]
+
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[tool.hatch.build.targets.wheel]
+packages = ["src/airflow_shared"]
```

```suggestion
packages = ["src/airflow_shared/timezones"]
```

or 

```suggestion
packages = ["src/airflow_shared", "src/airflow_shared/timzeones"]
```
I am not entirely sure what are the consequences of using the "packages" entry here in wheel ?

---

It didn't seem to affect anything and worked as it was? (I tested the `uv build shared/timezones` and then `unzip -l` to see what it put in it)

---

Ah indeed -> https://hatch.pypa.io/latest/config/build/#explicit-selection . This is only to aid explicit selection of what should be in the package, overriding default auto-detection. So "airflow_shared" in this case is enough (because there is nothing else than the right sub-package there):

```
Generic[¶](https://hatch.pypa.io/latest/config/build/#generic)
You can use the only-include option to prevent directory traversal starting at the project root and only select specific relative paths to directories or files. Using this option ignores any defined [include patterns](https://hatch.pypa.io/latest/config/build/#patterns).


[ pyproject.toml](https://hatch.pypa.io/latest/config/build/#pyprojecttoml_5)
[ hatch.toml](https://hatch.pypa.io/latest/config/build/#hatchtoml_4)

[tool.hatch.build.targets.sdist]
only-include = ["pkg", "tests/unit"]

Packages[¶](https://hatch.pypa.io/latest/config/build/#packages)
The packages option is semantically equivalent to only-include (which takes precedence) except that the shipped path will be collapsed to only include the final component.

So for example, if you want to ship a package foo that is stored in a directory src you would do:


[ pyproject.toml](https://hatch.pypa.io/latest/config/build/#pyprojecttoml_6)
[ hatch.toml](https://hatch.pypa.io/latest/config/build/#hatchtoml_5)

[tool.hatch.build.targets.wheel]
packages = ["src/foo"]
```